maven-plugin-tools-api/src/main/java/org/apache/maven/tools/plugin/util/PluginUtils.java

```diff
@@ -137,4 +145,50 @@ public int compare(Parameter parameter1, Parameter parameter2) {
             });
         }
     }
+
+    /**
+     * @param mojoClassName a fully qualified Mojo implementation class name, not null
+     * @param project a MavenProject instance, could be null
+     * @return <code>true</code> if the Mojo class implements <code>MavenReport</code>,
+     * <code>false</code> otherwise.
+     * @throws IllegalArgumentException if any
+     * @since 3.10.0
+     */
+    public static boolean isMavenReport(String mojoClassName, MavenProject project) throws IllegalArgumentException {
+        if (mojoClassName == null) {
+            throw new IllegalArgumentException("mojo implementation should be declared");
+        }
```

Comment:
A new method should use `Objects#notNull()` here and the caller should just wrap it.

Reply:
TBH the code was just moved from `GeneratorUtils` in maven-plugin-tools-generator in order to reduce dependencies. Let us improve the code in a separate commit/PR.

Reply:
Agreed. Please make a note.